package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +777,635 @@ def __setstate__(self, state):
+                    raise ValueError(f"{value} is not a unit recognizable by"
+                                     " MDAnalyis. Allowed units are:"
+                                     f" {self._unit_translation_dict.keys()}"
+                                     " For more information on units, see"
+                                     " `MDAnalyis units`_.")
+                else:
+                    self.units[key] = self._new_units[key]
+
+        if self.convert_units:
+            # check if all units are None
+            if not any(self.units.values()):
+                raise ValueError("The trajectory has no units, but "
+                                 "``convert_units`` is set to ``True`` by "
+                                 "default in MDAnalysis. To write the file "
+                                 "with no units, set ``convert_units=False``.")
+
+    def _open_file(self):
+        """Opens file with `H5PY`_ library and fills in metadata from kwargs.
+
+        :attr:`self.h5md_file` becomes file handle that links to root level.
+
+        """
+
+        if self._comm is not None:  # pragma: no cover
+            # ValueError raised if communicator is passed
+            # MPI parallel writing not yet supported
+            self.h5md_file = h5py.File(name=self.filename,
+                                       mode='w',
+                                       driver=self._driver,
+                                       comm=self._comm)
+        elif self._driver is not None:
+            self.h5md_file = h5py.File(name=self.filename,
+                                       mode='w',
+                                       driver=self._driver)
+        else:
+            self.h5md_file = h5py.File(name=self.filename,
+                                       mode='w')
+
+        # fill in H5MD metadata from kwargs
+        self.h5md_file.require_group('h5md')
+        self.h5md_file['h5md'].attrs['version'] = np.array(self.H5MD_VERSION)
+        self.h5md_file['h5md'].require_group('author')
+        self.h5md_file['h5md/author'].attrs['name'] = self.author
+        if self.author_email is not None:
+            self.h5md_file['h5md/author'].attrs['email'] = self.author_email
+        self.h5md_file['h5md'].require_group('creator')
+        self.h5md_file['h5md/creator'].attrs['name'] = self.creator
+        self.h5md_file['h5md/creator'].attrs['version'] = self.creator_version
+
+    def _initialize_hdf5_datasets(self, ts):
+        """initializes all datasets that will be written to by
+        :meth:`_write_next_timestep`
+
+        Note
+        ----
+        :exc:`NoDataError` is raised if no positions, velocities, or forces are
+        found in the input trajectory. While the H5MD standard allows for this
+        case, :class:`H5MDReader` cannot currently read files without at least
+        one of these three groups. A future change to both the reader and
+        writer will allow this case.
```

Comment:
Does not show up in docs. Summarize in main writer docs (and leave it here, too).